### PR TITLE
32684-filtered-mail-sender-implementation

### PR DESCRIPTION
**Pull request by Tom-Ritense**
-Implemented apply filters
-Fixed unit testing causing docker-compose startups

### CODE REVIEW

File: mail/src/main/kotlin/com/ritense/mail/service/FilteredMailSender.kt [hunk 1]
```diff
@@ -26,60 +26,63 @@ import java.util.Optional
...
-        var filteredRawMailMessage = rawMailMessage
-        val enabledMailFiltersSortedByPriority = getEnabledMailFiltersSortedByPriority()
-        for (mailFilter in enabledMailFiltersSortedByPriority) {
-            val optionalFilteredRawMailMessage = mailFilter.doFilter(filteredRawMailMessage)
-            filteredRawMailMessage = if (optionalFilteredRawMailMessage.isPresent) {
-                optionalFilteredRawMailMessage.get()
+    private fun applyFilters(rawMailMessage: RawMailMessage): Optional<RawMailMessage> {
+        var filteredRawMailMessage = rawMailMessage;
+        val filters = prioritizedFilters()
+        for (filter in filters) {
```
Comment (huibvanstraten) [inline mail/src/main/kotlin/com/ritense/mail/service/FilteredMailSender.kt:56]: .forEach{}?

File: mail/src/main/kotlin/com/ritense/mail/service/FilteredMailSender.kt [hunk 2]
```diff
@@ -26,60 +26,63 @@ import java.util.Optional
...
-        var filteredTemplatedMailMessage = templatedMailMessage
-        val enabledMailFiltersSortedByPriority = getEnabledMailFiltersSortedByPriority()
-        for (mailFilter in enabledMailFiltersSortedByPriority) {
-            val optionalFilteredTemplatedMailMessage = mailFilter.doFilter(filteredTemplatedMailMessage)
-            filteredTemplatedMailMessage = if (optionalFilteredTemplatedMailMessage.isPresent) {
-                optionalFilteredTemplatedMailMessage.get()
+    private fun applyFilters(templatedMailMessage: TemplatedMailMessage): Optional<TemplatedMailMessage> {
+        var filteredTemplatedMailMessage = templatedMailMessage;
+        val filters = prioritizedFilters()
+        for (filter in filters) {
```
Comment (huibvanstraten) [inline mail/src/main/kotlin/com/ritense/mail/service/FilteredMailSender.kt:70]: same here

**Review by huibvanstraten — APPROVED** [review]
Again just a small comment. Looks good. We are well on our way!

File: mail/src/main/kotlin/com/ritense/mail/service/FilteredMailSender.kt (hunk 1 above)
Comment (Tom-Ritense) [reply mail/src/main/kotlin/com/ritense/mail/service/FilteredMailSender.kt:56]: will do

File: mail/src/main/kotlin/com/ritense/mail/service/FilteredMailSender.kt (hunk 2 above)
Comment (Tom-Ritense) [reply mail/src/main/kotlin/com/ritense/mail/service/FilteredMailSender.kt:70]: will do

**Pull request merged**